docs/CORE_DEV_GUIDE.md

```diff
@@ -0,0 +1,180 @@
+
+5. **Implementations and algorithms:** You should understand the code being modified
+or added before approving it.  (See `Merge Only Changes You Understand` below.)
+Implementations should do what they claim, and be simple, readable, and efficient.
+
+6. **Tests:** All contributions *must* be tested, and each added line of code
+should be covered by at least one test. Good tests not only execute the code,
+but explores corner cases.  It can be tempting not to review tests, but please
+do so.
+
+Other changes may be *nitpicky*: spelling mistakes, formatting,
+etc. Do not insist contributors make these changes, and instead you can
+make the changes by [pushing to their branch](https://help.github.com/articles/committing-changes-to-a-pull-request-branch-created-from-a-fork/), or using GitHub’s [suggestion](https://help.github.com/articles/commenting-on-a-pull-request/) [feature]
+(https://help.github.com/articles/incorporating-feedback-in-your-pull-request/).
+(The latter is preferred because it gives the contributor a choice in
+whether to accept the changes.)
+
+Unless you know that a contributor is experienced with git, don’t
+ask for a rebase when merge conflicts arise. Instead, rebase the
+branch yourself, force-push to their branch, and advise the contributor to force-pull.  If the contributor is
+no longer active, you may take over their branch by submitting a new pull
+request and closing the original. In doing so, ensure you communicate
+that you are not throwing the contributor's work away!
+
+### Merge Only Changes You Understand
+
+*Long-term maintainability* is an important concern.  Code doesn't
+merely have to *work*, but should be *understood* by multiple core
+developers.  Changes will have to be made in the future, and the
+original contributor may have moved on.
+
+Therefore, *do not merge a code change unless you understand it*. Ask
+for help freely: we can consult community members, or even external developers,
+for added insight where needed, and see this as a great learning opportunity.
+
+While we collectively "own" any patches (and bugs!) that become part
+of the code base, you are vouching for changes you merge.  Please take
+that responsibility seriously.
+
+## Further resources
+
+As a core member, you should be familiar with community and developer
+resources such as:
+
+-  Our [contributor guide](CONTRIBUTING.md).
+-  Our [code of conduct](CODE_OF_CONDUCT.md).
+-  Our [governance](GOVERNANCE.md).
+-  Our [mission and values](MISSION_AND_VALUES.md).
+-  Our [benchmarking guide](BENCHMARKS.md).
+-  [PEP8](https://www.python.org/dev/peps/pep-0008/) for Python style.
+-  [PEP257](https://www.python.org/dev/peps/pep-0257/) and the
+   [NumPy documentation guide](https://docs.scipy.org/doc/numpy/docs/howto_document.html)
+   for docstrings. (NumPy docstrings are a superset of PEP257. You
+   should read both.)
+-  We use [`pre-commit`](https://pre-commit.com) hooks for autoformatting.
```

Comment:
I feel like the writing style of this changed from the above

Reply:
yeah, some has been copied / pasted from skimage, maybe worth standardizing a bit

Reply:
improved hopefully